src/Mod/CAM/PathSimulator/AppGL/GCodeParser.cpp

```diff
@@ -0,0 +1,234 @@
+/***************************************************************************
+ *   Copyright (c) 2024 Shai Seger <shaise at gmail>                       *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifdef _MSC_VER
+#ifndef _CRT_SECURE_NO_WARNINGS
+#define _CRT_SECURE_NO_WARNINGS
+#endif
+#endif
+
+#include "GCodeParser.h"
+#include <ctype.h>
+#include <stdio.h>
+
+using namespace MillSim;
+
+static char TokTypes[] = "GTXYZIJKR";
+
+GCodeParser::~GCodeParser()
+{
+    // Clear the vector
+    Operations.clear();
+}
+
+bool GCodeParser::Parse(const char* filename)
+{
+    Operations.clear();
+    lastState = {eNop, -1, 0, 0, 0, 0, 0, 0};
+    lastTool = -1;
+
+    FILE* fl;
```

Comment:
old hobbits of an old developer.  I will look for the cpp counterpart. This function is only used for my standalone version.

Reply:
It may be harder to implement using C++ APIs so if it is much work there is no need to push it. Profit from it is minimal but it is worth to try. Maybe freecad does already have something that would be helpful to you - I don't remember anything specific but I have not worked in that areas of code.

Reply:
We have `std::ifstream` with methods like `getline` or `eof`, it might be worth to try.